proto-grpc.md

```diff
@@ -0,0 +1,254 @@
+which we will explore later.
+
+The full list of available rulesets has been compiled in the table below, showing the languages
+supported by each. If you know of a ruleset not shown in this list, please open an issue
+[here](https://github.com/bazel-contrib/SIG-rules-authors/issues).
+
+<iframe src="https://docs.google.com/spreadsheets/d/e/2PACX-1vTiSt_QqCX1BHHf5BQZp5niRdBPwXbCn9TlpBwJrBRGEscZ75fQE8J4jupOMHu61uoAerUJsyXrCtDc/pubhtml?gid=0&amp;single=true&amp;widget=true&amp;headers=false" style="width: 100%; height: 400px"></iframe>
+
+
+## Picking a Ruleset
+
+Given the above options of rulesets, it may appear difficult to select which to use when introducing
+Protobuf or gRPC to a project. However, the selection essentially comes down to a process of
+elemination through three steps. As a prerequisite for these steps, you should consider the
+languages you currently need to support within your project and any you may expect to support in the
+near future, as this dictates the majority of the choices:
+
+1. In the event you are using only C++ and/or Python, consider using the rules provided in the
+   core dependency repositories above. Despite their lack of documentation, they generally work
+   acceptably for simple use-cases and benefit from not pulling in any further rule dependencies.
+   Simplicity is your friend in this case.
+
+2. If you have a single required language (or a low but manageable number), consider using the
+   language-specific rulesets if they are available and suitably documented. As mentioned before,
+   these rulesets benefit from being tailored to the language requirements and you will likely be
+   pulling these rulesets into your workspace already.
+
+3. If you have multiple supported languages or languages that do not have language-specific
+   proto/gRPC rules, consider using one of the two cross-language rulesets listed above. In this
+   case, you will benefit from the common rule interface and behaviour between your languages.
+   Choosing between the two rulesets will effectively come down to your preference for using
+   Gazelle within your project and the languages supported by each.
+
+
+## In Depth
+
+Given the above variety of options and rulesets, it is useful to explore why the ecosystem appears
+so fragmented and what makes this a more difficult problem that it may at first seem.
+
+At the top-level, all of these rulesets depend on the
+[bazelbuild/rules_proto](https://github.com/bazelbuild/rules_proto) ruleset described previously,
+which introduces the `proto_library` language-agnostic rule, which collects a set of `.proto` files
+and transforms them to a set of
+[descriptors](https://developers.google.com/protocol-buffers/docs/reference/cpp/google.protobuf.descriptor)
+and [ProtoInfo providers](https://docs.bazel.build/versions/main/skylark/lib/ProtoInfo.html)
+that other rules can use to drive protoc.
+
+However, the complications on top of this process come in three main varieties:
```

Comment:
there's a fourth variety, at least in JS, which is different wire formats (json vs binary) and different client library APIs (getter/setter vs. properties), and then when generating typings they may not 100% match the runtime library shape

Reply:
I've added a section about plugin configurability as a fourth point in that section